sync/sync.py

```diff
@@ -671,11 +674,24 @@ def finish(self, status="complete"):
             repo.git.worktree("prune")
 
     @mut()
-    def gecko_rebase(self, new_base):
-        new_base = sync_commit.GeckoCommit(self.git_gecko, new_base)
+    def gecko_rebase(self, new_base_ref, abort_on_fail=False):
+        new_base = sync_commit.GeckoCommit(self.git_gecko, new_base_ref)
         git_worktree = self.gecko_worktree.get()
-        git_worktree.git.rebase(new_base.sha1)
-        self.gecko_commits.base = new_base
+        set_new_base = True
+        try:
+            git_worktree.git.rebase(new_base.sha1)
+        except git.GitCommandError as e:
+            if abort_on_fail:
+                set_new_base = False
+                try:
+                    git_worktree.git.rebase(abort=True)
+                except git.GitCommandError:
+                    pass
```

Comment:
if the abort rebase fails, it just goes to raise an error. 
The message is ignored with the pass. why the `try` on `git_worktree.git.rebase(abort=True)`  then?

Reply:
The idea is that if we can't abort the rebase for whatever reason, we just want to raise the original error, since in either case we have a manual fixup to make, and it means the rest of the code just has a single error to handle. In practice the abort failing is rather unlikely (it probably means something more catastrophic is going on like the disk being full or similar).